src/plugins/intel_cpu/src/mkldnn_exec_network.cpp

```diff
@@ -239,6 +257,141 @@ InferenceEngine::Parameter MKLDNNExecNetwork::GetMetric(const std::string &name)
     }
 }
 
+bool MKLDNNExecNetwork::canBeExecViaLegacyDynBatch(std::shared_ptr<const ov::Model> function, int64_t& maxBatchSize) const {
+    maxBatchSize = -1;
+    auto isDynBatchWithUpperBound = [](const ov::PartialShape& shape) -> bool {
+        bool retVal = shape[0].is_dynamic() && shape[0].get_max_length() != -1;
+        for (size_t i = 1; i < shape.size(); i++) {
+            retVal = retVal && shape[i].is_static();
+        }
+        return retVal;
+    };
+
+    auto isValidParamsForDynBatchExec = [&maxBatchSize, &isDynBatchWithUpperBound](std::shared_ptr<const ov::Model> function) {
+        if (function->get_parameters().empty()) {
+            return false;
+        }
+
+        for (const auto& param : function->get_parameters()) {
+            const auto shape = param->get_output_partial_shape(0);
+            if (shape.rank().is_dynamic()) {
+                return false;
+            }
+
+            if (one_of(shape.rank().get_length(), 0, 1, 3)) {
```

Comment:
I took it from here: https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/cnn_network_ngraph_impl.cpp#L635

0 rank - means scalar and can't be dynamic at all.
We can't support dynamic shape with rank == 1 because primitives cant work with 1D dynamic shape like with dynamic batch.
But in fact we support rank 3, so removed from condition